lib/presentation/view/upcoming.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:carousel_slider/carousel_slider.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../../core/util/constants.dart';
+import '../../core/util/dimensions.dart';
+import '../../core/util/palette.dart';
+import '../../core/util/service_constants.dart';
+import '../../domain/entity/movie_event.dart';
+import '../bloc/interfaces/i_movies_bloc.dart';
+import '../widget/empty_widget.dart';
+import '../widget/error_widget.dart';
+import '../widget/show_image.dart';
+
+class Upcoming extends StatefulWidget {
+  const Upcoming({Key? key}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _UpcomingState();
+}
+
+class _UpcomingState extends State<Upcoming> {
+  @override
+  void initState() {
+    Provider.of<IMoviesBloc>(
+      context,
+      listen: false,
+    ).getMovies(ServiceConstants.endpoints['Upcoming']);
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: StreamBuilder(
+        stream: Provider.of<IMoviesBloc>(context).getStream(),
+        initialData: MovieEvent(status: Status.initial),
+        builder: (BuildContext context, AsyncSnapshot<MovieEvent> snapshot) {
+          switch (snapshot.data!.status) {
+            case Status.initial:
+              return const CircularProgressIndicator(
+                valueColor: AlwaysStoppedAnimation<Color>(Palette.primaryLight),
+              );
+            case Status.loading:
+              return const CircularProgressIndicator(
+                valueColor: AlwaysStoppedAnimation<Color>(Palette.primaryLight),
+              );
+            case Status.success:
+              return CarouselSlider.builder(
+                itemCount: snapshot.data!.movies!.length,
+                itemBuilder: (BuildContext context, int index, int idX) {
```

Comment:
trailing comma, use a more descriptive name to replace idX

Reply:
Done